### PR TITLE
Mount files from local HD

### PR DESCRIPTION
**Pull request by acedward**
https://github.com/user-attachments/assets/9396aa8c-4bb6-4ebb-92d2-6399dccf9c24

### CODE REVIEW

File: shinkai-libs/shinkai-http-api/src/api_v2/api_v2_handlers_app_files.rs [hunk 1]
```diff
@@ -0,0 +1,452 @@
...
+            Ok(warp::reply::with_status(warp::reply::json(&response), StatusCode::OK))
+        }
+        Err(error) => Ok(warp::reply::with_status(
+            warp::reply::json(&error),
+            StatusCode::from_u16(error.code).unwrap(),
+        )),
+    }
+}
+
+// #[derive(Deserialize, ToSchema)]
```
Comment (nicarq) [inline shinkai-libs/shinkai-http-api/src/api_v2/api_v2_handlers_app_files.rs:166]: delete?

File: shinkai-libs/shinkai-http-api/src/api_v2/api_v2_handlers_app_files.rs [hunk 2]
```diff
@@ -0,0 +1,452 @@
...
+// pub struct UpdateFileRequest {
+//     pub file_name: String,
+//     pub new_name: String,
+//     pub file_data: Vec<u8>,
+// }
+
+#[utoipa::path(
+    post,
+    path = "/v2/patch_app_file",
+    request_body = UpdateFileRequest,
```
Comment (nicarq) [inline shinkai-libs/shinkai-http-api/src/api_v2/api_v2_handlers_app_files.rs:176]: is this one correct

**Comment by nicarq** [comment]
besides those 2 minor comments, it looks good!

File: shinkai-libs/shinkai-http-api/src/api_v2/api_v2_handlers_app_files.rs [hunk 3]
```diff
@@ -0,0 +1,452 @@
...
+        )),
+    }
+}
+
+// #[derive(Deserialize, ToSchema)]
+// pub struct UpdateFileRequest {
+//     pub file_name: String,
+//     pub new_name: String,
+//     pub file_data: Vec<u8>,
+// }
```
Comment (acedward) [inline shinkai-libs/shinkai-http-api/src/api_v2/api_v2_handlers_app_files.rs:171]:
```suggestion
// #[derive(Deserialize, ToSchema)]
// pub struct UpdateFileRequest {
//     pub file_name: String,
//     pub new_name: String,
//     pub file_data: Vec<u8>,
// }
```

File: shinkai-libs/shinkai-http-api/src/api_v2/api_v2_handlers_app_files.rs (hunk 1 above)
Comment (acedward) [reply shinkai-libs/shinkai-http-api/src/api_v2/api_v2_handlers_app_files.rs]: Thanks for the catch. I changed from json to multi part and forgot to update the utopia

**Pull request merged**